tasks/install_dist_pkgs.yml

```diff
@@ -84,18 +84,46 @@
   args:
     warn: false
 
-- name: install agent
+- name: install agent pre2.8.0
+  package:
+    name: "newrelic-infra"
+    state: "{{ nrinfragent_state }}"
+  when: nrinfragent_os_name|lower != 'windows' and ansible_version.full < "2.8.0"
+
+- name: install agent post2.8.0
   package:
     name: "newrelic-infra"
     state: "{{ nrinfragent_state }}"
-  when: nrinfragent_os_name|lower != 'windows'
+  when: nrinfragent_os_name|lower != 'windows' and ansible_version.full >= "2.8.0" and nrinfragent_os_name|lower != 'amazon' and nrinfragent_os_name|lower != 'redhat'
 
-- name: install integrations
+- name: Install agent yum
+  yum:
+    name: "newrelic-infra"
+    state: "{{ nrinfragent_state }}"
+    lock_timeout: 30
+  when: nrinfragent_os_name|lower == 'redhat' and nrinfragent_os_name|lower == 'amazon' and ansible_version.full >= "2.8.0"
+
+- name: install integrations pre2.8.0
   package:
     name: "{{ item.name }}"
     state: "{{ item.state }}"
   with_items: "{{ nrinfragent_integrations }}"
-  when: nrinfragent_os_name|lower != 'windows'
+  when: nrinfragent_os_name|lower != 'windows' and ansible_version.full < "2.8.0"
+
+- name: Install integrations post2.8.0
+  package:
+    name: "{{ item.name }}"
+    state: "{{ item.state }}"
+  with_items: "{{ nrinfragent_integrations }}"
+  when: nrinfragent_os_name|lower != 'windows' and ansible_version.full >= "2.8.0" and nrinfragent_os_name|lower != 'amazon' and nrinfragent_os_name|lower != 'redhat'
+
+- name: Install integrations yum
+  yum:
+    name: "{{ item.name }}"
+    state: "{{ item.state }}"
+    lock_timeout: 30
```

Comment:
Sure thing, thanks for catching this oversight, I'll prepare a new PR resolving the issue for this task

Reply:
Hello! Any update on this?

Reply:
Hello, sorry for the delay. It has been added and released, so the lock_timeout should be present in the newest version (0.9.0) available at ansible-galaxy

Reply:
That's great, could you also fix the first task? We're still facing errors with that one.
```
---
- name: confirm redhat lsb util is present pre2.8.0
  yum:
    name: redhat-lsb-core
    state: present
    lock_timeout: "{{ nrinfragent_yum_lock_timeout }}" # add this line please
  when: nrinfragent_os_name|lower == 'redhat' 
```

Thanks a lot

Reply:
Ah in this case, there's something else going on here. In this particular instruction we can't use lock_timeout since this is an instruction for pre2.8.0 ansible. But it seems that we're missing a part of when here to ensure that it's used in versions lower than 2.8.0 so I will prepare a fix for this after we merge one of the bigger prs.

Reply:
I've added an issue for this section so the progress can be tracked here:
https://github.com/newrelic/infrastructure-agent-ansible/issues/119

Reply:
Oh I see, but in that case, maybe you need to check the ansible version like:
```
- name: confirm redhat lsb util is present pre2.8.0
  yum:
    name: redhat-lsb-core
    state: present
  when: nrinfragent_os_name|lower == 'redhat' and ansible_version.full < "2.8.0"
```
if not, you're running that task with all ansible version and nr infra agent = redhat
What do you think?

Reply:
That's exactly the case. I seem to have added it to the post2.8.0 version but not the pre2.8.0